Services/DeviceClient.cs

```diff
@@ -41,27 +42,54 @@ public class DeviceClient : IDeviceClient
         private readonly Azure.Devices.Client.DeviceClient client;
         private readonly ILogger log;
         private readonly IoTHubProtocol protocol;
+        private readonly IRateLimiting rateLimiting;
         private readonly string deviceId;
         private readonly IScriptInterpreter scriptInterpreter;
 
         //used to create method pointers for the device for the IoTHub to callback to
+        // ReSharper disable once NotAccessedField.Local
         private DeviceMethods deviceMethods;
 
+        private bool connected;
+
+        public IoTHubProtocol Protocol => this.protocol;
+
         public DeviceClient(
             Azure.Devices.Client.DeviceClient client,
             IoTHubProtocol protocol,
+            IRateLimiting rateLimiting,
             ILogger logger,
             string deviceId,
             IScriptInterpreter scriptInterpreter)
         {
             this.client = client;
             this.protocol = protocol;
+            this.rateLimiting = rateLimiting;
             this.log = logger;
             this.deviceId = deviceId;
             this.scriptInterpreter = scriptInterpreter;
+            this.connected = false;
         }
 
-        public IoTHubProtocol Protocol => this.protocol;
+        public async Task ConnectAsync()
+        {
+            if (this.client != null && !this.connected)
+            {
+                // TODO: HTTP clients don't "connect", find out how HTTP connections are measured and throttled
```

Comment:
GitHub Issue?

Reply:
added comment pointing to https://github.com/Azure/device-simulation-dotnet/issues/85